conans/test/functional/remote/multi_remote_test.py

```diff
@@ -95,6 +95,37 @@ def conan_install_build_flag_test(self):
         client_b.run("remote list_ref")
         self.assertIn("Hello0/0.0@lasote/stable: default", client_b.out)
 
+    def conan_install_update_test(self):
+        """
+        Checks conan install --update works only with the remote associated
+        """
+        client = TestClient(servers=self.servers, users={"default": [("lasote", "mypass")],
+                                                         "local": [("lasote", "mypass")]})
+
+        self._create(client, "Hello0", "0.0")
+        client.run("install Hello0/0.0@lasote/stable --build missing")
+        client.run("upload Hello0/0.0@lasote/stable --all -r default")
+        sleep(1)  # For timestamp and updates checks
+        self._create(client, "Hello0", "0.0", modifier=" ")
+        client.run("install Hello0/0.0@lasote/stable --build missing")
+        client.run("upload Hello0/0.0@lasote/stable --all -r local")
+        client.run("remove '*' -f")
+
+        client.run("install Hello0/0.0@lasote/stable")
+        self.assertIn("Hello0/0.0@lasote/stable from 'default' - Downloaded", client.out)
+        client.run("install Hello0/0.0@lasote/stable --update")
```

Comment:
Yes, seems the test is correct, validating current behavior. Could you please add the check that ``--update -r=local`` will actually update from local repo?